### ISSUE TITLE
Clicking 'Exit' in Network Options menu returns user to title screen instead of Options screen

### ISSUE THREAD

**Issue by Wallacoloo** [stepmania/stepmania#525]
Steps to reproduce (using Stepmania 5.0-git-de08067):
1. Starting from the main menu, click "Options".
2. Navigate to the "Network Options" item and hit enter.
3. Select the "Exit" option at the bottom of the list and hit enter.  You should now be taken back to the title screen.

In all the other options menus (Arcade Options, Appearance Options, etc), clicking "Exit" takes you back to the Options menu instead of to the title screen, so this behavior seems inconsistent.

**Comment by Wallacoloo** [comment]
This also happens on the "Select Game" options menu.

**Comment by kyzentun** [comment]
In the "Select Game" options menu, it is not a bug because Stepmania has to reload everything after the game type is changed.

**Comment by Wallacoloo** [comment]
Ok.

Regarding the network, I found this in [02 Branches.lua](https://github.com/stepmania/stepmania/blob/d35933f1e58205d7f95a36e34a7d1717095ae57f/Themes/_fallback/Scripts/02%20Branches.lua#L245):

```Lua
Network = function()
	return IsNetConnected() and "ScreenTitleMenu" or "ScreenTitleMenu"
end
```

This looks like a typo to me (as is, it's the equivalent of `return "ScreenTitleMenu"`), but I'm not too certain what the desired behavior is.

Is this another case where if the network was connected/disconnected everything must be reloaded?

**Comment by kyzentun** [comment]
My guess it that it's an interface design decision.  After someone successfully connects, they want to start playing, not waste time backing out of a menu.  (most videos of someone playing on keyboard have sped up animations, you can tell they're holding tab to get through screen transitions faster)
I'm not sure what is actually most useful to do when the connection failed or whatever.

**Comment by Wallacoloo** [comment]
In the appearance options menu, "Exit" normally takes you back to the the main options menu, but if the theme changed, then it takes you back to the title screen. Perhaps something like that could be applied to the "Select Game" menu as well?

The network settings would be a bit more difficult - I don't think I'll mess with those.

I was able to track down the line that is (indirectly) responsible for the conditional screen branching in the appearance options menu: https://github.com/stepmania/stepmania/blob/d35933f1e58205d7f95a36e34a7d1717095ae57f/src/ScreenOptionsMaster.cpp#L123

It looks like there's similar logic in place for detecting game changes already (line 142). Both Game::ChangeGame and Game::ChangeTheme will redirect to the InitialScreen if the game/theme was changed. So I think all that needs changing is 

```
NextScreen=Branch.TitleMenu()
```
to
```
NextScreen="ScreenOptionsService"
```
in metrics.ini.
